astropy/wcs/utils.py

```diff
@@ -1137,3 +1141,47 @@ def fit_wcs_from_points(xy, world_coords, proj_point='center',
                       np.zeros((degree+1, degree+1)), wcs.wcs.crpix)
 
     return wcs
+
+
+def obsgeo_to_frame(obsgeo, obstime):
+    """
+    Convert a WCS obsgeo property into an ITRS coordinate frame.
+
+    Parameters
+    ----------
+    obsgeo : np.ndarray
+        A shape ``(6, )`` array representing ``OBSGEO-[XYZ], OBSGEO-[BLH]`` as
+        returned by `astropy.wcs.WCS.wcs.obsgeo`.
+
+    obstime : time-like
+        The time assiociated with the coordinate, will be passed to
+        `astropy.coordinates.ITRS` as the obstime keyword.
+
+    Returns
+    -------
+    BaseCoordinateFrame
+        An `~astropy.coordinates.ITRS` coordinate frame representing the coordinates.
+    """
+
+    if (obsgeo is None
+        or np.all(obsgeo == 0)
+        or np.all(~np.isfinite(obsgeo))
```

Comment:
Shouldn't this be 
```suggestion
        or not np.all(np.isfinite(obsgeo))
```
??? For example, how would current logic work for `np.array([1, 1, np.nan, 1, 1, np.nan])`? I this is allowed to "pass" then the `if` below will not catch this `nan`:
```python
In [122]: a = np.array([1, 1, np.nan, 1, 1, np.nan])

In [123]: np.all(~np.isfinite(a))
Out[123]: False

In [124]: np.all(a[:3]  != 0)
Out[124]: True
```

Reply:
I did this like this to allow `np.array([1, 1, 1, np.nan, np.nan, np.nan])` through which would parse fine. I could modify the later conditionals to catch some NaNs. Unless someone wants there to be a NaN I guess??!